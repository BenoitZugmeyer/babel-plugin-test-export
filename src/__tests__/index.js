const { transform } = require("babel-core")

function transformCode(code) {
  return transform(code, { plugins: ["../../.."] }).code
}

test("simple function", () => {
  expect(
    transformCode(`
// @test-export
function foo() {}
    `)
  ).toMatchSnapshot()
})

test("simple variable", () => {
  expect(
    transformCode(`
// @test-export
const foo = 42;
    `)
  ).toMatchSnapshot()
})

test("function imbrication", () => {
  expect(
    transformCode(`
// @test-export
function foo(str) {
  console.log(str);
  function foo() {}
  foo();
}

// @test-export
function bar() {
  bar.biz = 1;
}

foo("foo");
    `)
  ).toMatchSnapshot()
})

test("variables usage inside functions", () => {
  expect(
    transformCode(`
// @test-export
let foo, bar;

// @test-export
let a = 42;

foo = 12;

bar = foo;
biz = 12;

function blah() {
  return foo;
}

function blih() {
  var foo;
  return foo;
}
    `)
  ).toMatchSnapshot()
})

test("previous statement trailing comment", () => {
  expect(
    transformCode(`
// foo
const Biz = {
    bar: true,
}

// @test-export
function format(rights) {
    console.log('whatever')
}

module.exports = {}
    `)
  ).toMatchSnapshot()
})

test("readme example", () => {
  expect(
    transformCode(`
// @test-export
function appendFoo(str) {
    return str + "foo"
}

module.exports = function (str) {
    return appendFoo(str).toUpperCase()
}
    `)
  ).toMatchSnapshot()
})
