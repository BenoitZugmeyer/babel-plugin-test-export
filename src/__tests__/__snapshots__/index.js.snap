// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function imbrication 1`] = `
"const __test__ = {};

// @test-export
__test__.foo = function foo(str) {
  console.log(str);
  function foo() {}
  foo();
}

// @test-export
;

__test__.bar = function bar() {
  __test__.bar.biz = 1;
};

__test__.foo.call(undefined, \\"foo\\");
module.exports.__test__ = __test__"
`;

exports[`previous statement trailing comment 1`] = `
"const __test__ = {};

// foo
const Biz = {
    bar: true

    // @test-export
};
__test__.format = function format(rights) {
    console.log('whatever');
};

module.exports = {};
module.exports.__test__ = __test__"
`;

exports[`readme example 1`] = `
"const __test__ = {};

// @test-export
__test__.appendFoo = function appendFoo(str) {
    return str + \\"foo\\";
};

module.exports = function (str) {
    return __test__.appendFoo.call(undefined, str).toUpperCase();
};
module.exports.__test__ = __test__"
`;

exports[`simple function 1`] = `
"const __test__ = {};

// @test-export
__test__.foo = function foo() {};

module.exports.__test__ = __test__"
`;

exports[`simple variable 1`] = `
"const __test__ = {};

// @test-export
__test__.foo = 42;
module.exports.__test__ = __test__"
`;

exports[`variables usage inside functions 1`] = `
"const __test__ = {};

// @test-export
__test__.foo = undefined;
__test__.bar = undefined;

// @test-export

__test__.a = 42;


__test__.foo = 12;

__test__.bar = __test__.foo;
biz = 12;

function blah() {
  return __test__.foo;
}

function blih() {
  var foo;
  return foo;
}
module.exports.__test__ = __test__"
`;
